File: apps/web/lib/passwordPolicy.ts
export const PASSWORD_MIN_LENGTH = 10;

export function validatePasswordStrength(password: string) {
  const value = password.trim();

  if (value.length < PASSWORD_MIN_LENGTH) {
    return `Use uma senha com pelo menos ${PASSWORD_MIN_LENGTH} caracteres.`;
  }

  if (!/[a-z]/.test(value)) {
    return "A senha precisa ter pelo menos uma letra minúscula.";
  }

  if (!/[A-Z]/.test(value)) {
    return "A senha precisa ter pelo menos uma letra maiúscula.";
  }

  if (!/\d/.test(value)) {
    return "A senha precisa ter pelo menos um número.";
  }

  if (!/[^A-Za-z0-9]/.test(value)) {
    return "A senha precisa ter pelo menos um símbolo.";
  }

  return null;
}

File: apps/mobile/lib/src/core/security/password_policy.dart
String? validatePasswordStrength(String password) {
  final value = password.trim();

  if (value.length < 10) {
    return 'Use uma senha com pelo menos 10 caracteres.';
  }

  if (!RegExp(r'[a-z]').hasMatch(value)) {
    return 'A senha precisa ter pelo menos uma letra minúscula.';
  }

  if (!RegExp(r'[A-Z]').hasMatch(value)) {
    return 'A senha precisa ter pelo menos uma letra maiúscula.';
  }

  if (!RegExp(r'\d').hasMatch(value)) {
    return 'A senha precisa ter pelo menos um número.';
  }

  if (!RegExp(r'[^A-Za-z0-9]').hasMatch(value)) {
    return 'A senha precisa ter pelo menos um símbolo.';
  }

  return null;
}
